code/datums/status_effects/buffs.dm

```diff
@@ -228,3 +228,30 @@
 	name = "Wish Granter's Immortality"
 	desc = "You are being resurrected!"
 	icon_state = "wish_granter"
+
+/datum/status_effect/cult_master
+	id = "The Cult Master"
+	duration = -1
+	tick_interval = 100
+	alert_type = null
+	var/alive = TRUE
+
+/datum/status_effect/cult_master/tick()
+	if(!owner)
+		src.Destroy()
+		return
+	if(owner.stat != DEAD && !alive)
+		alive = TRUE
+		return
+	if(owner.stat == DEAD && alive)
+		alive = FALSE
+		on_remove()
+
+/datum/status_effect/cult_master/on_remove()
```

Comment:
make what you're doing with this proc and call that proc in on_remove(), like below

    /datum/status_effect/cult_master/on_remove()
    remove_message_proc()
    . = ..()